src/main/scala/firrtl/transforms/BlackBoxSourceHelper.scala

```diff
@@ -35,6 +35,13 @@ case class BlackBoxPathAnno(target: ModuleName, path: String) extends BlackBoxHe
   override def serialize: String = s"path\n$path"
 }
 
+/** Exception indicating that a blackbox wasn't found
+  * @param fileName the name of the BlackBox file (only used for error message generation)
+  * @param e an underlying exception that generated this
+  */
+class BlackBoxNotFoundException(fileName: String, e: Throwable = null) extends FIRRTLException(
+  s"BlackBox '$fileName' not found. Did you misspell it? Is it in src/{main,test}/resources?", e)
```

Comment:
I may be biased in thinking that a user is doing test driven development. If you're running tests, e.g., in Chisel3, then anything in `src/test/resources` is okay and is fair game for being picked up. If you're not running tests, then I don't think this will get picked up. Do you think that this is confusing and should be removed?

Concretely, the following Chisel tests pass where `foo.v` is in src/test/resources, `bar.v` is in src/main/resources, and `baz.v` does not exist:
```scala
package chiselTests

import chisel3._
import chisel3.util.HasBlackBoxResource
import chisel3.experimental.MultiIOModule
import firrtl.transforms.BlackBoxNotFoundException

class Top(resource: String) extends MultiIOModule {
  val bb = Module(new BB(resource))
}

class BB(resource: String) extends BlackBox with HasBlackBoxResource {
  val io = IO(new Bundle {})
  setResource(resource)
}

class Sandbox extends ChiselFlatSpec {
  behavior of "Blackbox"

  it should "work since foo.v is in src/test/resources" in {
    Driver.execute(Array("-X", "verilog"), () => new Top("/foo.v"))
  }

  it should "work since bar.v is in src/main/resources" in {
    Driver.execute(Array("-X", "verilog"), () => new Top("/bar.v"))
  }

  it should "provide a good error message since baz.v doesn't exist" in {
    (the [BlackBoxNotFoundException] thrownBy { Driver.execute(Array("-X", "verilog"), () => new Top("/baz.v")) })
      .getMessage should include ("Is it in src/{main,test}/resources?")
  }
}
```

Then running this:
```
> find src -name \*.v && sbt "testOnly chiselTests.Sandbox"
src/main/resources/bar.v
src/test/resources/chisel3/VerilogVendingMachine.v
src/test/resources/chisel3/BlackBoxTest.v
src/test/resources/chisel3/AnalogBlackBox.v
src/test/resources/foo.v
[info] ...
[info] [0.000] Elaborating design...
[info] [0.004] Done elaborating.
[info] Sandbox:
[info] Blackbox
[info] - should work since foo.v is in src/test/resources
[info] - should work since bar.v is in src/main/resources
[info] - should provide a good error message since baz.v doesn't exist
[info] ScalaTest
[info] Run completed in 3 seconds, 584 milliseconds.
[info] Total number of tests run: 3
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 3, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 3, Failed 0, Errors 0, Passed 3
[success] Total time: 42 s, completed Oct 24, 2018 6:52:33 PM
```

Reply:
That's fair, yeah I was thinking more for people using chisel/firrtl in rocket-chip, but in general we encourage people to use ScalaTest for testing. I think you have the right error message.